util/postprocess-makedepend.pl

```diff
@@ -0,0 +1,160 @@
+            return ($target, $deps);
+        },
+    'VMS C' =>
+        sub {
+            my $line = shift;
+
+            # current versions of DEC / Compaq / HP / VSI C strips away all
+            # directory information from the object file, so we must insert it
+            # back. Just to be safe against future changes, we check that there
+            # really is no directory information.
+            my $directory = shift;
+
+            # The pattern for target and dependencies will always take this
+            # form:
+            #
+            #   target SPACE : SPACE deps
+            #
+            # This is so a volume delimiter (a : without any spaces around it)
+            # won't get mixed up with the target / deps delimiter.  We use this
+            # fact in the regexp below to make sure we do look at the target.
+            $line =~ s/^/$directory/ unless /^\S+[:>\]]\S+\s+:/;
+
+            # We know that VMS has system header files in text libraries,
+            # extension .TLB.  We also know that our header files aren't stored
+            # in text libraries.  Finally, we know that VMS C produces exactly
+            # one dependency per line, so we simply discard any line ending with
+            # .TLB.
+            return undef if /\.TLB\s*$/;
+
+            my ($target, $deps) = $line =~ /^(.*)\s:\s(.*)/;
+            $deps =~ s/^\s+//;
+            $deps =~ s/\s+$//;
+            return ($target, $deps);
+        },
+    'VC' =>
+        sub {
+            my $line = shift;
+            my $object = shift;
+
+            # For the moment, we only support Visual C on native Windows, or
+            # compatible compilers.  With those, the flags /Zs /showIncludes
+            # give us the necessary output to be able to create dependencies
+            # that nmake (or any 'make' implementation) should be able to read,
+            # with a bit of help.  The output we're interested in looks like
+            # this (it always starts the same)
+            #
+            #   Note: including file: {whatever header file}
+            #
+            # So all we really have to do is to is to replace the start of the
+            # line with an object file specification, given to us as an extra
+            # argument (passed from $ARGV[1]);
+            #
+            # There are also other lines mixed in, for example compiler
+            # warnings, so we simply discard anything that doesn't start with
+            # the Note:
+
+            if (/^Note: including file: */) {
+                (my $tail = $') =~ s/\s*\R$//;
+
+                # VC gives us absolute paths for all include files, so to
+                # remove system header dependencies, we need to check that
+                # they don't match $abs_srcdir or $abs_blddir
+                $tail = canonpath($tail);
+                if ($tail =~ m|^\Q$abs_srcdir\E|i
+                        || $tail =~ m|^\Q$abs_blddir\E|i) {
```

Comment:
I'd like to suggest [one last thing]
```
if ($tail =~ s|^\Q$abs_blddir\E|\$(BLDDIR)|i || tail =~ s|^\Q$abs_srcdir\E|$(SRCDIR)|i)
```
Yes, there is ambiguity when you build in source directory, but I'd say it worths it. In case you wonder BLDDIR being moved forward is not a coincidence, but way to make it work when you build in subdirectory of the source directory.

Reply:
There is internal knowledged on what files are being generated (i.e. should be in `$(BLDDIR)`) and which ones are not (i.e. should be in `$(SRCDIR)`)...  so it wouldn't be *extremely* hard to get those right.

Reply:
In such case note that this can just as well be said about makedepend [and presumably VMS]. However! I'd like to suggest to *not* take that path, because it might turn out more expensive than the value it adds. "More expensive" means that it would have to take in information about *all* files or per-file basis. "Per-file" means you execute this script for each file you compile.

Reply:
I'd say that if you choose to fold/reduce/canonize pathnames in dependency lists, then add-depends.pl would probably be better place.

Reply:
This is surely worth looking at for VMS as well...  but not the use of `$(SRCDIR)` and `$(BLDDIR)`, as directory merging isn't as simple as on Unix or Windows...

Regarding the information, this script is already sucking in `configdata.pm`, and `keys %{$unified_info{generated}}` is all the information you need...

Reply:
> I'd say that if you choose to fold/reduce/canonize pathnames in dependency lists, then add-depends.pl would probably be better place.

Good point, performance wise.

Reply:
I assume that includes what you suggested here, btw.

Reply:
> I assume that includes what you suggested here, btw.

I don't understand this sentence. Do you mean that you suggest to leave it as it is here and now and do "canonization" in add-depends.pl on all platforms? It would do it, but since I didn't quite get it, could you be more explicit?

Reply:
As for sucking in configdata.pm and performance. One has to recognize that it's easy to fall victim to illusion that resources are unlimited. But it doesn't change the fact that it's still an illusion. One just adds lines of code without much of a consideration, and then one day you step back and say [what happened?](https://github.com/openssl/openssl/issues/5585). In this context one can actually wonder if it would be appropriate to even split configdata.pm to something minimalistic and say fileinfo.pm. Or somehow arrange that the hash that lists files is not actually instantiated unless you ask for it...

Reply:
> I don't understand this sentence. Do you mean that you suggest to leave it as it is here and now and do "canonization" in add-depends.pl on all platforms? It would do it, but since I didn't quite get it, could you be more explicit?

At the beginning of this comment thread, you suggest this change:

```
if ($tail =~ s|^\Q$abs_blddir\E|\$(BLDDIR)|i || tail =~ s|^\Q$abs_srcdir\E|$(SRCDIR)|i)
```

I assume that this would also do better in `add-depends.pl`

Reply:
> In this context one can actually wonder if it would be appropriate to even split configdata.pm to something minimalistic and say fileinfo.pm.

It might be a bit late in the game to refactor at that level, at least for 1.1.1

> Or somehow arrange that the hash that lists files is not actually instantiated unless you ask for it...

Soooo....  you're saying that I should qualify the use of `configdata` a bit?  Like this?

``` perl
use configdata qw(%config);
```

Reply:
> I assume that this would also do better in add-depends.pl.

Yes. Under assumption that you actually venture into file canonization in add-depends.pl. Because I don't feel that one actually has to. And if one doesn't (or chooses to do it later), then suggested replacement would do some good. At least it would be a bit more consistent with Unix, which is customarily free from absolute paths.

Reply:
> use configdata qw(%config);

I suppose. I'm not that much into Perl internals, but if this doesn't instantiate a big fat hash, then yes, it would be totally appropriate.

Reply:
> Under assumption that you actually venture into file canonization in add-depends.pl. Because I don't feel that one actually has to.

*If* we're doing that, I would rather have it done in *one* place.

> (or chooses to do it later)

"later", is that "post 1.1.1"?  I mean, fixing such things in `add-depend.pl` can certainly be done before the final release...

> > ``` perl
> > use configdata qw(%config);
> 
> I suppose. I'm not that much into Perl internals, but if this doesn't instantiate a big fat hash, then yes, it would be totally appropriate.

Nah, I realised that makes no real difference...  It will still instanciate everything from that file, the LIST following the module name is just about what's getting imported into the current module (main::).  It's possible that perl will choose to free the unused stuff when it sees that nothing refers to, but...

Reply:
> If we're doing that, I would rather have it done in one place.

Sure, and one should be able to revert this replacement when/if you collect yourself to actually doing it [in that one magic place].

Reply:
Side note: Actually, it just dawned on me that we can as well do all this post-processing stuff in `add-depends.pl`

Reply:
Yeah. Nicest things would be that you'd do only one canonization per header file [and then reuse the result in each subsequent occurrence.]

Reply:
Ah, a result cache. That can be done.

Reply:
So what's final conclusion? Do you seek approval as it is, or do you choose to keep the suggested replacement till better times?

Reply:
For the moment, I'm away from the computer... but I'm pondering moving this all to `add-depends.pl` and work from there.

Reply:
I rather meant what's final conclusion for *this* request. It feels that add-depends.pl thing is matter for separate request...

Reply:
For this request, I'm done